src/pad.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+import {Tensor} from './lib/tensor.js';
+
+/**
+ * Get mapped location from source tensor.
+ * @param {Array} location
+ * @param {Array} inputShape
+ * @param {Array} beginningPadding
+ * @param {MLPaddingMode} mode
+ * @return {Array} mapped location of source tensor
+ */
+function getMappedLocation(location, inputShape, beginningPadding, mode) {
+  const mappedLocation = location.slice();
+  const rank = location.length;
+  if (mode === 'edge') {
+    for (let i = 0; i < rank; i++) {
+      if (location[i] < beginningPadding[i]) {
+        mappedLocation[i] = 0;
+      } else if (location[i] >= beginningPadding[i] + inputShape[i]) {
+        mappedLocation[i] = inputShape[i] - 1;
+      } else {
+        mappedLocation[i] -= beginningPadding[i];
+      }
+    }
+  } else {
+    // reflection mode or symmetric mode
+    const offset = mode === 'symmetric' ? 1 : 0;
+    for (let i = 0; i < rank; i++) {
+      if (mappedLocation[i] < beginningPadding[i]) {
+        // mappedLocation[i] =
+        //    beginningPadding[i] + (beginningPadding[i] - mappedLocation[i]) -
+        //    beginningPadding[i] - offset;
+        mappedLocation[i] = beginningPadding[i] - mappedLocation[i] - offset;
+      } else if (mappedLocation[i] >= beginningPadding[i] + inputShape[i]) {
+        // mappedLocation[i] =
+        //    beginningPadding[i] + inputShape[i] - 1 -
+        //    (mappedLocation[i] - (beginningPadding[i] + inputShape[i] -1)) -
+        //    beginningPadding[i] + offset;
```

Comment:
Are these comments just showing the full equation before cancelling out terms? At first I thought it was old code that was simply forgotten to be deleted.

How about :
```
       // Full equation:
       // mappedLocation[i] =
       //    beginningPadding[i] + (beginningPadding[i] - mappedLocation[i]) -
       //    beginningPadding[i] - offset;
```

Reply:
@fdwr Yes, these comments are full equation for understanding easily. 
Updated by new commit, thanks.

Reply:
Why not just use the full equation in the code? I suppose it would erase the confusion and won't require the comment.